### PR TITLE
Update MUI X

### PR DESCRIPTION
**Pull request by maheichyk**
MUI X date and time pickers are updated.

**:heavy_check_mark: Checklist**

- [x] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [x] Added or updated documentation.
- [x] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: matrix-meetings-widget/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "@mui/lab": "^5.0.0-alpha.133",
     "@mui/material": "^5.13.4",
     "@mui/utils": "^5.13.1",
-    "@mui/x-date-pickers": "^5.0.20",
+    "@mui/x-date-pickers": "^6.8.0",
```
Comment (dhenneke) [inline matrix-meetings-widget/package.json:21]: We can already update to [`6.10.2`](https://github.com/mui/mui-x/releases)

**Review by dhenneke — COMMENTED** [review]
Looks good already

File: matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx [hunk 2]
```diff
@@ -103,11 +78,23 @@ export function EndDatePicker({
...
+          'aria-label': openDatePickerDialogue(value),
+        },
+        textField: {
+          fullWidth: true,
+          helperText:
+            (!date.isValid() &&
+              t('dateTimePickers.invalidEndDate', 'Invalid date')) ||
+            (typeof error === 'string' ? error : undefined),
+          margin: 'dense',
+          error: !date.isValid() || !!error || undefined,
```
[the remark this answers is not in the document]
Comment (maheichyk) [reply matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx:95]: This is something that seems not to be mentioned in migration guide or APIs, was found during implementation.

File: .changeset/blue-laws-brush.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@nordeck/matrix-meetings-widget': minor
+---
+
+Update MUI X date and time pickers.
```
[the remark this answers is not in the document]
Comment (maheichyk) [reply .changeset/blue-laws-brush.md]: I like the idea to write about changed time picker in the changeset.

File: matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.test.tsx [hunk 4]
```diff
@@ -110,8 +115,8 @@ describe('<EndDatePicker/>', () => {
     // userEvent.type doesn't work here, so we have to use fireEvent
     fireEvent.change(textbox, { target: { value: '99/99/9999' } });
 
-    expect(textbox).toHaveAccessibleDescription('Invalid date');
-    expect(textbox).toBeInvalid();
+    expect(textbox).not.toHaveAccessibleDescription('Invalid date');
```
[the remark this answers is not in the document]
Comment (maheichyk) [reply matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.test.tsx]: Looks good to me, thanks for suggestion.

File: matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx [hunk 5]
```diff
@@ -103,11 +78,23 @@ export function EndDatePicker({
...
       disablePast={!enablePast}
-      getOpenDialogAriaText={getOpenDialogAriaText}
       label={t('dateTimePickers.endDate', 'End date')}
       onChange={handleOnChange}
       reduceAnimations={isReduceAnimations()}
-      renderInput={renderInput}
+      slotProps={{
+        openPickerButton: {
+          'aria-label': openDatePickerDialogue(value),
+        },
```
[the remark this answers is not in the document]
Comment (maheichyk) [reply matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx:87]: Sounds good to me. Let's change this code when new mui-x version is released.

**Review by dhenneke — APPROVED** [review]

File: matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dhenneke) [reply matrix-meetings-widget/src/components/common/DateTimePickers/EndDatePicker.tsx:87]: -> https://github.com/nordeck/matrix-meetings/pull/244

**Pull request merged**